src/gf.c

```diff
@@ -2004,7 +2004,7 @@ JL_DLLEXPORT void jl_method_instance_add_backedge(jl_method_instance_t *callee,
                 if (ciedge != (jl_value_t*)caller)
                     continue;
                 jl_value_t *invokeTypes = i > 0 ? jl_array_ptr_ref(backedges, i - 1) : NULL;
-                if (invokeTypes && jl_is_method_instance(invokeTypes))
```

Comment:
Seems like you meant for it to have both (neither is deprecated, they just have different meanings)

Reply:
Huh, the implementation of `get_next_edge` strongly suggests the caller is always a `CodeInstance`:
```c
int get_next_edge(jl_array_t *list, int i, jl_value_t** invokesig, jl_code_instance_t **caller)
```